### DISCUSSION TITLE
Play mp3 files discord.js

### DISCUSSION THREAD

**Discussion by TimoCoupek** [discordjs/discord.js#6064] [Q&A] [answered]
The bot enters and does nothing does not even give an error the piece of code to run the file is as follows:

var channel = message.guild.channels.cache.get("842859974324518913")
if (!channel) return console.error("The channel does not exist!");
channel.join().then(connection => {
    const dispatcher = connection.play('./audios/private.mp3');
    dispatcher.on("end", end => {
        channel.leave();
    });
});
and i dont understand beaucose, can help me everyone

**Comment by kyranet** [comment, marked as answer]
discord.js v12's voice is not longer maintained and does not receive any updates for bugfixes or any other. Please consider using [`@discordjs/voice`](https://github.com/discordjs/voice), which is a newer voice module for discord.js with better stability and performance.

**Reply by TimoCoupek** [reply]
Sorry I hope I'm not asking too much, but should I change the code structure?

**Reply by kyranet** [reply]
You'll need to change quite a lot of your voice-related structure with the new voice library, yes.

You can get more support and information in the `#djs-new-voice` channel from the [Discord server](https://discord.gg/djs).
